### PR TITLE
@remotion/noise

### PR DESCRIPTION
**Pull request by satelllte**
## About

- Resolves: https://github.com/remotion-dev/remotion/issues/1395

## New docs pages

- https://remotion-git-fork-satelllte-noise-remotion.vercel.app/docs/add-noise
- https://remotion-git-fork-satelllte-noise-remotion.vercel.app/docs/noise
- https://remotion-git-fork-satelllte-noise-remotion.vercel.app/docs/noise/create-noise-2d
- https://remotion-git-fork-satelllte-noise-remotion.vercel.app/docs/noise/create-noise-3d
- https://remotion-git-fork-satelllte-noise-remotion.vercel.app/docs/noise/create-noise-4d

### CODE REVIEW

File: packages/docs/docs/noise/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: "@remotion/noise"
+---
+
+<!-- TODO: most probably, the version will be different, so change it later -->
+_Available from v3.2.31_
```
Comment (satelllte) [inline packages/docs/docs/noise/index.md]: @JonnyBurger you can change it before merging

**Comment by satelllte** [comment]
Hi, @JonnyBurger 

I think it's ready for review. Hope I didn't miss anything when creating the new package.

Thanks ! 🙏

File: packages/noise/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  createNoise3D as _createNoise3D,
+  createNoise4D as _createNoise4D,
+} from 'simplex-noise';
+import type {
+  NoiseFunction2D,
+  NoiseFunction3D,
+  NoiseFunction4D,
+} from 'simplex-noise';
+
+type RemotionSeed = number | string | null;
```
Comment (satelllte) [inline packages/noise/src/index.ts]: @JonnyBurger I think it worth to consider making the `RemotionSeed` type available to export from `remotion`

Comment (satelllte) [reply packages/noise/src/index.ts]: Logged an issue: https://github.com/remotion-dev/remotion/issues/1402

**Comment by JonnyBurger** [comment]
Awesome, you nailed even the details! 🙌  Quality work!
I'm planning on doing the final polishing tomorrow and am also considering to simplify the API:

```diff
- const noise = createNoise2D('my-seed')(px, frame)
+ const noise = noise2D('my-seed', px, frame)
```

And cache the expensive initialization via WeakMap.

**Comment by satelllte** [comment]
Thanks @JonnyBurger. I'm glad you liked it. 
Oh, I see what you mean. I din't pay the attention to the function signature initially.
Do you need me to tweak it? 

Also, I'd actually consider to leave both implementations:

In first case scenario, user wants to use it on the go:

```ts
const p1 = noise2D('seed-1', x, y);
const p2 = noise2D('seed-2', x, y);
```

But on the other hand I think it would be helpful to have the ability to just create the "instanced" noise function which doesn't require `seed` argument no more:

```ts
const myNoise = createNoise2D('my-seed');
const p1 = myNoise(x1, y1);
const p2 = myNoise(x2, y2);
```

**Comment by JonnyBurger** [comment]
@satelllte No action required on your side I think, I'll confirm that by tomorrow and get it shipped myself!

I like it more if there is just one way of doing it, also to make it easier to teach in the future. I prefer the pure functional way over the two-step and think that sometimes the seed is redundant is  an okay tradeoff. But if you strongly believe I should not change it, I have an open ear.

**Comment by JonnyBurger** [comment]
Thanks for understanding! Tried to simplify the example as well, to hopefully make it easy to grasp what noise is all about. I see you contributed to simplex-noise as well, this is awesome!

Ready to ship 🚀  Thanks a lot for this super excellent PR 🙌

**Comment by JonnyBurger** [comment]
@satelllte Do you remember how long you spent on this issue?

BTW, going to also make a Twitter post about it an honor your packages once I have time to prepare all the announcements 😄

**Comment by satelllte** [comment]
> @satelllte Do you remember how long you spent on this issue?
> 
> BTW, going to also make a Twitter post about it an honor your packages once I have time to prepare all the announcements 😄

@JonnyBurger I think it's about a day or so. Cool, thank you!

**Pull request merged**